bundles/org.openhab.binding.etherrain/src/main/java/org/openhab/binding/etherrain/internal/handler/EtherRainHandler.java

```diff
@@ -0,0 +1,239 @@
+        }
+
+        switch (response.getOperatingStatus()) {
+            case STATUS_READY:
+                updateState(EtherRainBindingConstants.CHANNEL_ID_OPERATING_STATUS, new StringType("READY"));
+                break;
+            case STATUS_WAITING:
+                updateState(EtherRainBindingConstants.CHANNEL_ID_OPERATING_STATUS, new StringType("WAITING"));
+                break;
+            case STATUS_BUSY:
+                updateState(EtherRainBindingConstants.CHANNEL_ID_OPERATING_STATUS, new StringType("BUSY"));
+                break;
+        }
+
+        switch (response.getLastCommandStatus()) {
+            case STATUS_OK:
+                updateState(EtherRainBindingConstants.CHANNEL_ID_COMMAND_STATUS, new StringType("OK"));
+                break;
+            case STATUS_ERROR:
+                updateState(EtherRainBindingConstants.CHANNEL_ID_COMMAND_STATUS, new StringType("ERROR"));
+                break;
+            case STATUS_UNATHORIZED:
+                updateState(EtherRainBindingConstants.CHANNEL_ID_COMMAND_STATUS, new StringType("UNATHORIZED"));
+                break;
+        }
+
+        switch (response.getLastCommandResult()) {
+            case RESULT_OK:
+                updateState(EtherRainBindingConstants.CHANNEL_ID_OPERATING_RESULT, new StringType("OK"));
+                break;
+            case RESULT_INTERRUPTED_RAIN:
+                updateState(EtherRainBindingConstants.CHANNEL_ID_OPERATING_RESULT, new StringType("RAIN INTERRUPTED"));
+                break;
+            case RESULT_INTERUPPTED_SHORT:
+                updateState(EtherRainBindingConstants.CHANNEL_ID_OPERATING_RESULT,
+                        new StringType("INTERRUPPTED SHORT"));
+                break;
+            case RESULT_INCOMPLETE:
+                updateState(EtherRainBindingConstants.CHANNEL_ID_OPERATING_RESULT, new StringType("DID NOT COMPLETE"));
+                break;
+        }
+
+        updateState(EtherRainBindingConstants.CHANNEL_ID_RELAY_INDEX, new DecimalType(response.getLastActiveValue()));
+
+        OnOffType rs = OnOffType.OFF;
+
+        if (response.isRainSensor()) {
+            rs = OnOffType.ON;
+        }
+
+        updateState(EtherRainBindingConstants.CHANNEL_ID_SENSOR_RAIN, rs);
+
+        logger.debug("Completed Etherrain Update");
+
+        return true;
+    }
+
+    private boolean execute() {
+        if (device != null) {
+            device.commandIrrigate(config.programDelay, EtherRainConfiguration.zoneOnTime1,
+                    EtherRainConfiguration.zoneOnTime2, EtherRainConfiguration.zoneOnTime3,
+                    EtherRainConfiguration.zoneOnTime4, EtherRainConfiguration.zoneOnTime5,
+                    EtherRainConfiguration.zoneOnTime6, EtherRainConfiguration.zoneOnTime7,
+                    EtherRainConfiguration.zoneOnTime8);
+            updateBridge();
+        }
+
+        return true;
+    }
+
+    private boolean clear() {
+        if (device != null) {
```

Comment:
Just deal with the null warnings:
```suggestion
        EtherRainCommunication device = this.device;
        if (device != null) {
```